browser/src/Editor/SimpleNeovimEditor.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * NeovimEditor.ts
+ *
+ * IEditor implementation for Neovim
+ */
+
+import * as React from "react"
+
+import { IncrementalDeltaRegionTracker } from "./../DeltaRegionTracker"
+import { NeovimInstance } from "./../neovim"
+import { CanvasRenderer, DOMRenderer, INeovimRenderer } from "./../Renderer"
+import { NeovimScreen } from "./../Screen"
+
+import * as Config from "./../Config"
+
+// import { PluginManager } from "./../Plugins/PluginManager"
+
+import { Keyboard } from "./../Input/Keyboard"
+import { IEditor } from "./Editor"
+
+import { NeovimRenderer } from "./NeovimRenderer"
+
+export class DummyPluginManager {
+    public getAllRuntimePaths(): string[] {
+        return []
+    }
+
+    public startPlugins(neovimInstance: NeovimInstance): void {
+        // no-op
+    }
+}
+
+export class SimpleNeovimEditor implements IEditor {
+
+    private _neovimInstance: NeovimInstance
+    private _deltaRegionManager: IncrementalDeltaRegionTracker
+    private _renderer: INeovimRenderer
+    private _screen: NeovimScreen
+
+    private _pendingTimeout: any = null
+    private _pendingAnimationFrame: boolean = false
+
+    constructor(
+        private _config: Config.Config = Config.instance(),
+    ) {
+        // TODO: How to get rid of this?
+        this._neovimInstance = new NeovimInstance(new DummyPluginManager(), 100, 100)
+        this._neovimInstance.setInitVim("C:/oni/test.vim")
+        this._deltaRegionManager = new IncrementalDeltaRegionTracker()
+        this._screen = new NeovimScreen(this._deltaRegionManager)
+
+        this._renderer = this._config.getValue("editor.renderer") === "canvas" ? new CanvasRenderer() : new DOMRenderer()
+
+        this._render()
+
+        this._config.registerListener(() => this._onConfigChanged())
+
+        this._neovimInstance.on("action", (action: any) => {
+            this._renderer.onAction(action)
+            this._screen.dispatch(action)
+
+            this._scheduleRender()
+
+            if (!this._pendingTimeout) {
+                this._pendingTimeout = setTimeout(() => this._onUpdate(), 0)
+            }
+        })
+
+        const keyboard = new Keyboard()
```

Comment:
having our own events that plugins can listen to sound great.
above with the other editors, we can have on load event with what kind of editor abstraction it is.

Also if we change the filetype of it to be something specific, we can have autocmds in vim for users to interact as well if they have something they would like to do in vim when a folder opens.

Also I know a lot of people have the need/want to open the folder pane on load to whichever side they want on open, this is something we need to think about as well.

Thanks for listening to my ideas, Oni just keeps more and more in line with what I was thinking for an editor.